### PR TITLE
mv nightly dist tests

### PR DESCRIPTION
**Pull request by sljlp**
### PR types

Others
### PR changes

Others
### Describe

1. fix doc in readme
2. move nightly dist tests to collective
3. update tool scripts to keep dist ports stable

### CODE REVIEW

**Review by XieYunshen — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/collective/README.md [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
 * `arch`: the device's architecture. similar to `os`, multiple valuse ars splited by ";" and ignoring case. The supported architectures are `gpu`, `xpu`, `ASCEND`, `ASCEND_CL` and `rocm`.
-* `timeout`: timeout of a unittest, whose unit is second.
+* `timeout`: timeout of a unittest, whose unit is second. Blank means defalut.
 * `run_type`: run_type of a unittest. Supported values are `NIGHTLY`, `EXCLUSIVE`, `CINN`, `DIST`, `GPUPS`, `INFER`, `EXCLUSIVE:NIGHTLY`, `DIST:NIGHTLY`，which are case-insensitive. 
-* `launcher`: the test launcher.Supported values are test_runner.py, dist_test.sh and custom scripts' name.
-* `num_port`: the number os port used in a distributed unit test
-* `run_serial`: whether in serial mode. the value can be 1 or 0.Default (empty) is 0.
+* `launcher`: the test launcher.Supported values are test_runner.py, dist_test.sh and custom scripts' name. Blank means test_runner.py.
+* `num_port`: the number of port used in a distributed unit test. Blank means no auto distributed port. 
+* `run_serial`: whether in serial mode. the value can be 1 or 0.Default (empty) is 0. Blank means defalut.
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/collective/README.md:15]: Wonderful！

File: tools/gen_ut_cmakelists.py [hunk 2]
```diff
@@ -247,24 +264,122 @@ def parse_line(line, curdir):
...
+    '''
+    Find all signed ut ports in cmake_file and update the ASSIGNED_PORTS
+    and keep the DIST_UT_PORT max of all assigned ports 
+    '''
+    with open(cmake_file_name) as cmake_file:
+        port_reg = re.compile("PADDLE_DIST_UT_PORT=[0-9]+")
+        lines = cmake_file.readlines()
+        for idx, line in enumerate(lines):
+            matched = port_reg.search(line)
+            if matched is not None:
```
Comment (gongweibao) [inline tools/gen_ut_cmakelists.py]:
```
if matched is None:
       continue
```

**Review by gongweibao — CHANGES_REQUESTED** [review]
1. Be aware of your code style
2. Write full description of you PR

File: tools/gen_ut_cmakelists.py (hunk 2 above)
Comment (sljlp) [reply tools/gen_ut_cmakelists.py]: done

File: tools/gen_ut_cmakelists.py [hunk 3]
```diff
@@ -161,146 +162,317 @@ def process_name(name, curdir):
...
+            current_work_dir = "."
+
+        contents = os.listdir(current_work_dir)
+        contents.sort()
+        sub_dirs = []
+        for c in contents:
+            c_path = os.path.join(current_work_dir, c)
+            if c_path in self.PROCESSED_DIR:
+                return
+            if os.path.isdir(c_path):
```
Comment (gongweibao) [inline tools/gen_ut_cmakelists.py]:
```
if not 
     continue
```

File: tools/gen_ut_cmakelists.py [hunk 4]
```diff
@@ -161,146 +162,317 @@ def process_name(name, curdir):
...
+
+
+class CMakeGenerator():
+
+    def __init__(self, root_dirs):
+        self.PROCESSED_DIR = set()
+        self.port_manager = DistUTPortManager()
+        self.root_dirs = root_dirs
+
+    def prepare_dist_ut_port(self):
```
Comment (gongweibao) [inline tools/gen_ut_cmakelists.py:336]: Add comments  why get parent instead of down.

File: tools/gen_ut_cmakelists.py [hunk 5]
```diff
@@ -161,146 +162,317 @@ def process_name(name, curdir):
...
-    global DIST_UT_PORT
-    port = DIST_UT_PORT
-    assert port < 23000, "dist port is exahausted"
-    DIST_UT_PORT += int(port_num)
-    return port
-
+class DistUTPortManager():
+
+    def __init__(self):
+        self.DIST_UT_PORT = 21200
```
Comment (gongweibao) [inline tools/gen_ut_cmakelists.py]: `DIST_UT_PORT`

**Review by gongweibao — CHANGES_REQUESTED** [review]
Some comments.

File: tools/gen_ut_cmakelists.py (hunk 5 above)
Comment (sljlp) [reply tools/gen_ut_cmakelists.py]: changed

File: tools/gen_ut_cmakelists.py (hunk 4 above)
Comment (sljlp) [reply tools/gen_ut_cmakelists.py:336]: done

File: tools/gen_ut_cmakelists.py [hunk 6]
```diff
@@ -161,146 +162,330 @@ def process_name(name, curdir):
...
+            self._gen_cmakelists(c)
+
+    def _find_root_dirs(self):
+        root_dirs = []
+        # for each current directory, find its highest ancient directory (at least itself)
+        # which includes CMakeLists.txt or testslist.csv.txt in the filesys tree
+        for c in self.current_dirs:
+            c = os.path.abspath(c)
+            while True:
+                ppath = os.path.dirname(c)
```
Comment (gongweibao) [inline tools/gen_ut_cmakelists.py:353]: https://stackoverflow.com/questions/2860153/how-do-i-get-the-parent-directory-in-python

Comment (gongweibao) [reply tools/gen_ut_cmakelists.py:353]: https://www.geeksforgeeks.org/get-parent-of-current-directory-using-python/

**Review by gongweibao — CHANGES_REQUESTED** [review]

File: tools/gen_ut_cmakelists.py (hunk 3 above)
Comment (sljlp) [reply tools/gen_ut_cmakelists.py]: done

**Review by XieYunshen — APPROVED** [review]

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**